scripts/python_codegen.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 SECO Mind Srl
+# SPDX-License-Identifier: Apache-2.0
+
+set -eEuo pipefail
+
+codegen () {
+    PROTO_DIR=$1
+    PROJECT_DIR=$2
+    OUT_DIR=$3
+    DL_DIR=$4
+
+    # Remove old code
+    if [ -d "$OUT_DIR/astarteplatform" ]; then
+        rm -r "$OUT_DIR/astarteplatform"
+    fi
+
+    if [ ! -d "$DL_DIR/grpc" ]; then
+        cd "$DL_DIR"
+        git clone -b v1.58.1 https://github.com/grpc/grpc
+        cd grpc
+        git submodule update --init
+    fi
+
+    if [ ! -f "$DL_DIR/grpc/grpc_python_plugin" ]; then
+        cd "$DL_DIR/grpc"
+        cmake .
+        make -j$(nproc) grpc_python_plugin
+        python3 -m pip install --upgrade pip
+        python3 -m pip install termcolor
+        cd -
+    fi
```

Comment:
This should probably be another Makefile step maybe, to not download the dependencies every time

Reply:
I think should be in another PR